python2/koans/about_asserts.py

```diff
@@ -53,7 +53,7 @@ def test_that_unittest_asserts_work_the_same_way_as_python_asserts(self):
         """
 
         # This throws an AssertionError exception
-        assert False
+        self.assertFalse
```

Comment:
Two things:

First, here `assertFalse` is not being executed, just evaluated. To see the difference open a python terminal and create a function:

```
>>>def fun():
...    print("hello world")
...
>>>fun()
hello world
>>> fun
<function fun at 0x7fdc722dc5f0>
>>>
```

Second, assertFalse requires parameters: https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertFalse . It takes a parameter that needs to evaluate to `False` or an exception will be raised.

I don't think you actually need to use `assertFalse` here, but just to change the assertion so that it doesn't throw anymore. The lesson here is that the mechanism through which assert works is using exceptions, and in particular a `AssertionError` exceptions. `assertFalse` and the other assertions from the UnitTest module do exactly the same, but they also add debug information which cannot be provided when using the plain `assert` from the Python standard (a failure message, for instance).

Look at: https://stackoverflow.com/questions/2958169/what-are-the-advantages-or-difference-in-assert-false-and-self-assertfalse

Reply:
Addressed